tests/test_class.cpp

```diff
@@ -21,6 +21,10 @@ struct NoBraceInitialization {
     std::vector<int> vec;
 };
 
+struct NotRegistered {};
+struct StringWrapper { std::string str; };
+NotRegistered test_error_after_conversions(StringWrapper) { return {}; }
+
```

Comment:
These classes should be moved with the module definitions, and the function rewritten as a lambda, unless there's a good reason to keep them outside.  (We try to enforce that style in the test suite, where possible, to keep the test cases as localized as possible within each test suite file, to makes it easier to read).